clifford/__init__.py

```diff
@@ -979,44 +979,9 @@ def __init__(self, layout, value=None, string=None):
                     "value must be a sequence of length %s" %
                     self.layout.gaDims)
 
-    def __array_wrap__(self, out_arr, context=None):
-        '''
-        This is a work-around needed to prevent numpy arrays from
-        operating  on MultiVectors using their operators.  This happens
-        because Multivectors act enough like an array, that duck
-        typing thinks they are an array.
-        '''
-        
-        uf, objs, out_index = context
-        op0 = objs[0]
-
-        # If its an exp it only has one argument
-        if uf.__name__ == 'exp':
-            return math.e ** (op0)
-
-        op1 = objs[1]
-
-        # Check if one of them subclasses from ndarray and the other doesn't
-        op0flag = issubclass(np.ndarray, op0)
-        op1flag = issubclass(np.ndarray, op1)
-        if (not op0flag) and op1flag:
-            op0 = np.empty(1, dtype=object)
-            op0[0] = objs[0]
-        elif (not op1flag) and op0flag:
-            op1 = np.empty(1, dtype=object)
-            op1[0] = objs[1]
-
-        if uf.__name__ =='multiply':
-            return op1*op0
-        if uf.__name__ == 'divide':
-            return op1.inv()*op0
-        elif uf.__name__ == 'add':
-            return op1 + op0
-        elif uf.__name__ == 'subtract':
-            return -op1 + op0
-
-        else:
-            raise ValueError('i dont know how to %s yet' % uf.__name__)
+    def __array__(self):
+        # we are a scalar, and the only appropriate dtype is an object array
+        return MVArray([self])
```

Comment:
This is somewhat sloppy, ideally you'd return a 0d array here with `MVArray(self)`. Looking at `MVArray` though, that doesn't look supported right now.